app/js/lib/foxbox/network.js

```diff
@@ -187,94 +180,76 @@ export default class Network extends EventDispatcher {
   }
 
   /**
-   * Ping the local box to detect whether it's still alive.
+   * Pings both local and tunnel box links simultaneously (if discovered).
    *
-   * @return {Promise}
    * @private
    */
-  [p.pingLocalBox]() {
-    // @todo Find a better way to detect if a local connection is active.
-    if (!this[p.settings].localOrigin) {
-      return Promise.resolve();
+  [p.pingLinks]() {
+    if (this[p.localLink]) {
+      this[p.localLink].ping();
     }
 
-    return this[p.ping](`${this.localOrigin}/ping`)
-      .then((isOnline) => this[p.onPong](p.local, isOnline));
+    if (this[p.tunnelLink]) {
+      this[p.tunnelLink].ping();
+    }
   }
 
   /**
-   * Ping the remote box to detect whether it's still alive.
-   *
-   * @return {Promise}
+   * Updates specified link. Link is updated if origin has changed, if origin
+   * became "null" link will be deleted.
+   * 
+   * @param {Symbol} symbol Symbol associated either with local or tunnel link.
+   * @param {string?} origin Origin string, if "null" - origin is not available
+   * anymore.
    * @private
    */
-  [p.pingRemoteBox]() {
-    // @todo Find a better way to detect if a tunnel connection is active.
-    if (!this[p.settings].tunnelOrigin) {
-      return Promise.resolve();
+  [p.updateLink](symbol, origin) {
+    const link = this[symbol];
```

Comment:
P.S. Not really the code I'm proud of, but don't have anything better in my mind for the moment :/